src/server/rest/v1/service/AuthorizationService.ts

```diff
@@ -33,16 +32,25 @@ import _ from 'lodash';
 const MODULE_NAME = 'AuthorizationService';
 
 export default class AuthorizationService {
-  public static canPerfomAction(entity: AuthorizationActions, authAction: Action): boolean {
-    switch (authAction) {
+  public static canPerfomAction(authActions: AuthorizationActions, action: Action): boolean {
+    const siteAreaAuthActions = <SiteAreaAuthorizationActions> authActions;
```

Comment:
As wrote in my comments already:

Use in the signature of the method:
`canPerfomAction(authActions: AuthorizationActions | SiteAreaAuthorizationActions, ...)`

In the code you downcast like this (by memory 😊):

```
case Action.ASSIGN_CHARGING_STATIONS:
  return (siteAreaAuthActions as SiteAreaAuthorizationActions).canAssignChargingStations
```